game_event/tests.py

```diff
@@ -265,3 +296,53 @@ def test_fail_add_player_to_event_already_in_it(self, client, saved_game_event,
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestWeatherApi:
+    def test_one_valid_response(self, client, saved_game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+
+        client.force_login(player.user)
+
+        output = get_weather_from_api([saved_game_event], mock_urlopen)
+        assert output[saved_game_event.id] == {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
```

Comment:
I think the test will be more complete if we assert on the complete output rather than one value (imagine a regression in which the function injects a trailing entry - you won't catch it if you only look at a specific entry):
```suggestion
        expected_weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
        assert output == {
                saved_game_event.id: expected_weather
        }
```

Reply:
@yftacherzog Can you please elaborate on this? 

> (imagine a regression in which the function injects a trailing entry

I'm not sure I understand the scenario.

Reply:
Your function generates an output which is a dictionary. Currently, you're assuming that it contains only the entry you expect to be generated. It might be that it's generating some other entries as well. For that reason, it's better to assert on the entire output rather than specific portions of it - when possible.

Suppose that you expect the output to be:
```python
{
  "x": 5
}
```
You test it with:
```python
assert output["x"] == 5
```

Tomorrow I make a change that introduces a regression that makes it return
```python
{
  "x": 5,
  "super-secret-stuff": "super-secret",
}
```
Your test still passes.

I suggest you change the test so it asserts the complete output:
```python
assert output == {"x": 5}
```

Reply:
Got it. thank you.
Will be fixed @yftacherzog

Reply:
@yftacherzog @IdanZimi Fixed